### PR TITLE
Biogen locator no longer has an unscoped locate

### PR DESCRIPTION
**Pull request by optimumtact**
Relies on PR #43792 merge that first

also WIP untested

### CODE REVIEW

**Comment by AnturK** [comment]
Locate gone but this is still spoofable.

**Comment by optimumtact** [comment]
how is it spoofable

**Comment by AnturK** [comment]
create=xraygun;amount=10

**Comment by optimumtact** [comment]
oh I forgot to make sure it was in the stored research, easy enough to fix

File: code/modules/hydroponics/biogenerator.dm [hunk 1]
```diff
@@ -309,8 +309,21 @@
...
+			stack_trace("ID did not map to a researched datum [id]")
+			return
+
+		//Get design by id (or may return error design)
+		var/datum/design/D = SSresearch.techweb_design_by_id(id)
+		//Valid design datum, amount and the datum is not the error design, lets proceed
+		if(D && amount && !istype(D, /datum/design/error_design))
+			create_product(D, amount)
+		//This shouldnt happen normally but href forgery is real
+		stack_trace("ID could not be turned into a valid techweb design datum [id]")
```
Comment (Menshin) [inline code/modules/hydroponics/biogenerator.dm]:
```suggestion
		else
		    stack_trace("ID could not be turned into a valid techweb design datum [id]")
```

**Pull request merged**